plugins/modules/foreman_templates_import.py

```diff
@@ -0,0 +1,185 @@
+    required: false
+    type: str
+  associate:
+    description: Associate to OSes, Locations and Organizations based on metadata.
+    required: false
+    type: str
+    choices:
+     - always
+     - new
+     - never
+  verbose:
+    description: Add template diffs to the output.
+    required: false
+    type: bool
+  force:
+    description: Update templates that are locked.
+    required: false
+    type: bool
+  lock:
+    description: Lock imported templates.
+    required: false
+    type: bool
+  branch:
+    description: Branch in Git repo.
+    required: false
+    type: str
+  repo:
+    description: Filesystem path or repo (with protocol), for example /tmp/dir or git://example.com/repo.git or https://example.com/repo.git.
+    required: false
+    type: str
+  filter:
+    description:
+      - Sync only templates with name matching this regular expression, after I(prefix) was applied.
+      - Case-insensitive, snippets are not filtered.
+    required: false
+    type: str
+  negate:
+    description: Negate the prefix.
+    required: false
+    type: bool
+  dirname:
+    description: The directory within Git repo containing the templates.
+    required: false
+    type: str
+  locations:
+    description: REPLACE locations with given list.
+    required: false
+    type: list
+    elements: str
+  organizations:
+    description: REPLACE organizations with given list.
+    required: false
+    type: list
+    elements: str
+extends_documentation_fragment:
+  - foreman
+'''
+
+EXAMPLES = '''
+- name: Sync templates from git repo
+  foreman_templates_import:
+    repo: https://github.com/theforeman/community-templates.git
+    branch: 1.24-stable
+    associate: new
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "changeme"
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanAnsibleModule
+
+
+def main():
+    module = ForemanAnsibleModule(
+        foreman_spec=dict(
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            associate=dict(choices=['always', 'new', 'never']),
+            prefix=dict(),
+            branch=dict(),
+            repo=dict(),
+            filter=dict(),
+            dirname=dict(),
+            verbose=dict(type='bool'),
+            force=dict(type='bool'),
+            lock=dict(type='bool'),
+            negate=dict(type='bool'),
+            locations=dict(type='entity_list', flat_name='location_ids'),
+            organizations=dict(type='entity_list', flat_name='organization_ids'),
+        ),
+        supports_check_mode=False,
+    )
+
+    module_params = module.clean_params()
+    module.connect()
+
+    if 'template' in module.foremanapi.resources or 'templates' in module.foremanapi.resources:
+        resource_name = 'template'
```

Comment:
shouldn't it be `resource_name = 'templates'` in the second case?

Reply:
oops, true

Reply:
-- this can be required_plugins=[('templates', ['*'])] in __init__ these days
but will it handle the rename of template resource to templates in the newer foreman?

Reply:
@evgeni, Ok, I see, this can be moved to required_plugins (adding the plugin to _PLUGIN_RESOURCES and a little kludge for template/templates somewhere along the lines), but I still need to know which resource to call the resource_action with.

Reply:
Yeah, now that you said that, `required_plugins` does not cope with that weird renaming well :(
So maybe, in this case, let the kludge be here and don't bother with `required_plugins` for now?